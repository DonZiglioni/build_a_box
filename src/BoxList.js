import React, { useState } from "react";
import Box from "./Box";
import BoxForm from "./BoxForm";
import './BoxList.css'

const BoxList = () => {
    const INITIAL_STATE = [];
    const [boxes, setBoxes] = useState(INITIAL_STATE);

    //  **  Adds a new Box to the BoxList  **
    const addBox = (id, color, height, width) => {
        setBoxes(boxes => [...boxes, { id, color, height, width }]);
    };

    //  **  Removes a Box from the BoxList  **
    const removeBox = (id) => {
        setBoxes(boxes.filter(box => box.id !== id));
    };

    //  **  Returns a BoxList and the BoxForm  **
    return (
        <>
            <h1>Build a Box</h1>
            <BoxForm addBox={addBox} />
            <div className="box-list">
                {boxes.map(({ id, color, height, width }) =>
                    <Box
                        id={id}
                        key={id}
                        color={color}
                        height={height}
                        width={width}
                        removeBox={removeBox}
                    />
                )}
            </div>
        </>
    )
}

export default BoxList;