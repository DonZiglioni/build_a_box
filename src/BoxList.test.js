import React from "react";
import { render } from "@testing-library/react"
import BoxList from "./BoxList";

it('Renders Successfully', () => {
    render(<BoxList />);
})

it('Matches Snapshot', () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
})