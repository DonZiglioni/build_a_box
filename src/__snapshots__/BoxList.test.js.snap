// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<DocumentFragment>
  <h1>
    Build a Box
  </h1>
  <form>
    <label
      for="color"
    >
      <b>
        Color: 
      </b>
       
      <i>
        #000000
      </i>
      <input
        id="color"
        name="color"
        type="color"
        value="#000000"
      />
    </label>
    <div
      class="ranges"
    >
      <div
        class="rangeH"
      >
        <label
          for="height"
        >
          <u>
            Height
          </u>
        </label>
        <p>
          50
        </p>
        <input
          id="height"
          max="150"
          min="50"
          name="height"
          step="10"
          type="range"
          value="50"
        />
      </div>
      <div
        class="rangeW"
      >
        <label
          for="width"
        >
          <u>
            Width
          </u>
        </label>
        <p>
          50
        </p>
        <input
          id="width"
          max="150"
          min="50"
          name="width"
          step="10"
          type="range"
          value="50"
        />
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    class="box-list"
  />
</DocumentFragment>
`;
